crates/sui-analytics-indexer/src/handlers/mod.rs

```diff
@@ -153,3 +156,20 @@ impl ObjectStatusTracker {
         }
     }
 }
+
+async fn get_move_struct(
+    struct_tag: &StructTag,
+    contents: &[u8],
+    package_cache: &PackageCache,
+) -> Result<MoveStruct> {
+    let move_struct = match package_cache
```

Comment:
@amnn would this be the right way to convert a sui object to `MoveStruct` while using package cache?

Reply:
This looks right to me! (But of course, the real measure of correctness is whether tests that involve both types defined in original and updated packages pass).